pkg/async-gateway/service.go

```diff
@@ -52,25 +55,42 @@ func NewService(clusterUID, apiName string, queue Queue, storage Storage, logger
 }
 
 // CreateWorkload enqueues an async workload request and uploads the request payload to S3
-func (s *service) CreateWorkload(id string, payload io.Reader, contentType string) (string, error) {
+func (s *service) CreateWorkload(id string, payload io.Reader, headers http.Header) (string, error) {
+	contentType := headers.Get("Content-Type")
+	if contentType == "" {
```

Comment:
Since the `Content-Type` header is checked twice for its presence, what do you think of removing it from the calling function?

Reply:
*Assuming the `Content-Type` will still be around*